security/authorization/util/util_test.go

```diff
@@ -0,0 +1,127 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package engine
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/internal/grpctest"
+
+	cel "github.com/google/cel-go/cel"
+	decls "github.com/google/cel-go/checker/decls"
+	expr "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+func (s) TestStringConvert(t *testing.T) {
+	declarations := []*expr.Decl{
+		decls.NewIdent("request.url_path", decls.String, nil),
+		decls.NewIdent("request.host", decls.String, nil),
+		decls.NewIdent("request.method", decls.String, nil),
+		decls.NewIdent("request.headers", decls.NewMapType(decls.String, decls.String), nil),
+		decls.NewIdent("source.address", decls.String, nil),
+		decls.NewIdent("source.port", decls.Int, nil),
+		decls.NewIdent("source.principal", decls.String, nil),
+		decls.NewIdent("destination.address", decls.String, nil),
+		decls.NewIdent("destination.port", decls.Int, nil),
+		decls.NewIdent("connection.uri_san_peer_certificate", decls.String, nil)}
+	env, _ := cel.NewEnv() // External Evaluation env does not have to be built the same
+	for _, test := range []struct {
+		desc                      string
+		expectedEvaluationOutcome bool
+		expectedParsingError      bool
+		expectedEvaluationError   bool
+		expr                      string
+		authzArgs                 map[string]interface{}
+	}{
+		{
+			desc:                      "1: use a policy with startsWith primitive and returns a match.",
```

Comment:
Please skim through the `Running specific tests or benchmarks` section in https://blog.golang.org/subtests to understand how subTest names are generated. Prefer using something which is more readable when the spaces are replaced with underscores.